### ISSUE TITLE
Add support for image previews of JPEG files in insecure http mode

### ISSUE THREAD

**Issue by jmikedupont2** [coder/code-server#7082]
What is your suggestion?

Add support for image previews of JPEG files when served via the VS Code web server, allowing users to view image content directly in the browser without needing to download or open the files separately.

---

Why do you want this feature?

This feature would improve user experience by enabling quick visualization of image assets directly in the web server interface, which is particularly useful for web developers, designers, and anyone working with visual media. It would save time and streamline workflows, especially when dealing with multiple images.

---

Are there any workarounds to get this functionality today?

Currently, users can manually open JPEG files in a browser tab or use third-party tools to preview images. However, these workarounds disrupt the development workflow and are less efficient compared to having built-in preview support.

---

Are you interested in submitting a PR for this?

Potentially, yes, it would be a great learning experience
![Image](https://github.com/user-attachments/assets/dfa7719c-d08f-451c-a5ed-f7e618eaeefe)

**Comment by code-asher** [comment]
This does actually work, but I think it might require a secure context because it relies on some browser features only available in secure contexts.  Are you able to use https or add an exception for `http://192.168.1.90:8080` in your browser?

**Comment by jmikedupont2** [comment]
i guess i can do that with a self signed https, will try, but not the best user experience for n00bs

**Comment by code-asher** [comment]
Yeah, it would be handy if we could support http, but VS Code relies on service workers for web views, and browsers only allow those in a secure context, so our hands are tied.  I think this could possibly be changed upstream in VS Code, but trying to patch it here is out of scope I think.

**Comment by code-asher** [comment]
If you do try to submit a PR to VS Code, I think the reason this happens is because image previews use web views, and web views use service workers, which browsers only allow in secure contexts.  So to make it work in http the preview would have to no longer use service workers, which may not be possible because they are used to provide isolation and it might be a security risk.  Still, for the image preview case maybe it is possible to use a simpler version that does not need any service workers.  Not sure though, some investigation would be required.
